nucypher/blockchain/eth/agents.py

```diff
@@ -1719,18 +1719,20 @@ def get_agent(cls,
         if not issubclass(agent_class, EthereumContractAgent):
             raise TypeError(f"Only agent subclasses can be used from the agency.")
 
+        registry_id = None
         if not registry:
             if len(cls.__agents) == 1:
-                _registry_id = list(cls.__agents.keys()).pop()
+                registry_id = list(cls.__agents.keys()).pop()
             else:
                 raise ValueError("Need to specify a registry in order to get an agent from the ContractAgency")
 
+        registry_id = registry_id or registry.id
```

Comment:
Minor thing, but I'd go for 
```
else:
    registry_id = registry.id
```
(and remove the `registry_id = None` line).

Reply:
:heavy_check_mark: